modules/reindex/src/test/java/org/opensearch/index/reindex/MultiCodecReindexTests.java

```diff
@@ -0,0 +1,160 @@
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .put("index.codec", randomFrom(codecChoices))
+                    .put(MergePolicyProvider.INDEX_MERGE_ENABLED, false)
+                    .build()
+            );
+            ensureGreen(index);
+
+            // index using all codecs
+            for (String codec : codecMap.keySet()) {
+                useCodec(index, codec);
+                ingestDocs(index, nbDocs);
+            }
+
+            assertTrue(
+                getSegments(index).stream()
+                    .flatMap(s -> s.getAttributes().values().stream())
+                    .collect(Collectors.toSet())
+                    .containsAll(codecMap.values())
+            );
+
+            // create destination index with destination codec
+            createIndex(
+                destIndex,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .put("index.codec", destCodec)
+                    .build()
+            );
+            ensureGreen(destIndex);
+
+            // perform reindex
+            BulkByScrollResponse response = reindex().source(index)
+                .destination(destIndex)
+                .refresh(true)
+                .waitForActiveShards(ActiveShardCount.ONE)
+                .get();
+            final int expectedResponseSize = codecMap.size() * nbDocs;
+
+            // assertions
+            assertEquals(0, response.getNoops());
+            assertEquals(1, response.getBatches());
+            assertEquals(0, response.getDeleted());
+            assertEquals(0, response.getVersionConflicts());
+            assertEquals(0, response.getBulkFailures().size());
+            assertEquals(0, response.getSearchFailures().size());
+
+            assertEquals(expectedResponseSize, response.getTotal());
+            assertEquals(expectedResponseSize, response.getCreated());
+
+            assertTrue(response.getTook().getMillis() > 0);
+            assertTrue(getSegments(destIndex).stream().allMatch(segment -> segment.attributes.containsValue(destCodecMode)));
+        }
+    }
+
+    private void useCodec(String index, String codec) throws ExecutionException, InterruptedException {
+        assertAcked(client().admin().indices().prepareClose(index).setWaitForActiveShards(1));
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .updateSettings(new UpdateSettingsRequest(index).settings(Settings.builder().put("index.codec", codec)))
+                .get()
+        );
+
+        assertAcked(client().admin().indices().prepareOpen(index).setWaitForActiveShards(1));
+    }
+
+    private void ingestDocs(String index, int nbDocs) throws InterruptedException {
+        indexRandom(
+            false,
```

Comment:
those used to be random flags, why we changed them?

Reply:
I was reducing randomness while trying to debug offline. Seems like I didn't revert back. I'll post another revision reverting the original behavior.

Also, `codecov/patch` is unhappy. Do we need to add extra coverage as well?

Reply:
> Also, `codecov/patch` is unhappy. Do we need to add extra coverage as well?

Codecov check could be skipped (for this pull request) I believe (since we essentially moving test case)